src/simplecov.jl

```diff
@@ -15,39 +15,67 @@ Simple(w::AbstractWeights; corrected::Bool = false) = Simple(corrected, w)
 
 
 """
-    cov(x::AbstractVector, sc::Simple)
-
-Compute the sample variance of the vector `x`. The sum is scaled with `n`
-where `n = length(x)` if `sc.corrected` is false and with `n-1` otherwise.
-"""
-cov(x::AbstractVector, sc::Simple) = cov(x; corrected = sc.corrected)
-
-
-"""
-    cov(x::AbstractVector, y, sc::Simple)
+    cov(x::AbstractVector, y::AbstractVector, sc::Simple)
 
 Compute the covariance of the vectors `x` and `y` using formula
 ``\\frac{1}{n}\\sum_{i=1}^n (x_i-\\bar x) (y_i-\\bar y)^*`` where ``*`` denotes
 complex conjugate. If `sc.corrected` is true then the fraction ``\\frac{1}{n}``
 is replaced with ``\\frac{1}{n-1}``.
 """
-cov(x::AbstractVector, y::AbstractVector, sc::Simple) = cov(x, y; corrected = sc.corrected)
+cov(x::AbstractVector, y::AbstractVector, sc::Simple) =
+    cov(x, y; corrected=sc.corrected)
 
 
 """
-    cov(X::AbstractMatrix, sc::Simple; dims::Int=1)
+    cov(X::AbstractMatrix, sc::Simple; dims=1, mean=nothing)
 
-Compute the covariance matrix associated with `X` along dimension `dims`.
-The sum is scaled with `n` where  `n = length(x)` if `sc.corrected` is false
-and with `n-1` otherwise.
+Compute the sample variance of `X`. The sum is scaled with the number of
+observations `n` if `sc.corrected` is false and with `n-1` otherwise.
+If `dims=1` the rows are assumed to be observations and the columns the
+features. If `dims=2` the opposite is assumed.
 """
-cov(X::AbstractMatrix, sc::Simple; dims::Int=1) = cov(X; dims=dims, corrected = sc.corrected)
+function cov(X::AbstractMatrix, sc::Simple; dims::Int=1, mean=nothing)
+    @assert dims ∈ [1, 2] "Argument dims can only be 1 or 2 (given: $dims)"
+
+    # weighted case via StatsBase (no argument mean supported)
+    sc.weights === nothing || return StatsBase.cov(X, sc.weights, dims;
+                                        corrected=sc.corrected)
+    # unweighted case via Statistics
+    if mean === nothing
+        return Statistics.cov(X; dims=dims, corrected=sc.corrected)
+    elseif iszero(mean)
+        return Statistics.covzm(X, dims, corrected=sc.corrected)
+    elseif mean isa AbstractArray
```

Comment:
so `mean(randn(5, 7), dims=1)` will not return a vector but an `AbstractArray{2, T}`. We could flatten it or write our own mean, in fact I think `_vmean` in `StatsBase` or maybe in `Statistics` one or the other, is a hand cooked one so that it is a vector. I'll look into improving this